subsys/net/lib/conn_mgr/conn_mgr.c

```diff
@@ -30,62 +30,70 @@ K_SEM_DEFINE(conn_mgr_event_signal, 1, 1);
 /* Used to protect conn_mgr state */
 K_MUTEX_DEFINE(conn_mgr_lock);
 
-static enum conn_mgr_state conn_mgr_iface_status(int index)
+static bool conn_mgr_iface_is_oper_up(int index)
 {
-	if (iface_states[index] & CMGR_IF_STATE_UP) {
-		return CMGR_STATE_CONNECTED;
-	}
-
-	return CMGR_STATE_DISCONNECTED;
+	return (iface_states[index] & CMGR_IF_STATE_UP) != 0;
 }
 
 #if defined(CONFIG_NET_IPV6)
-static enum conn_mgr_state conn_mgr_ipv6_status(int index)
+static bool conn_mgr_ipv6_status(int index)
 {
-	if ((iface_states[index] & CONN_MGR_IPV6_STATUS_MASK) ==
-	    CONN_MGR_IPV6_STATUS_MASK) {
+	bool ready = (iface_states[index] & CONN_MGR_IPV6_STATUS_MASK) == CONN_MGR_IPV6_STATUS_MASK;
```

Comment:
why this variable? just keep the code as it was and return directly true/false instead of CMGR_STATE_CONNECTE/CMGR_STATE_DISCONNECTED